### PR TITLE
Fix missing pip issue on travis

### PR DESCRIPTION
**Pull request by dorukozturk**

### CODE REVIEW

**Review by nicain — DISMISSED** [review]
Looks good!

**Comment by dorukozturk** [comment]
@nicain This will take a while (1 hour or so). Once travis passes I will merge it and you can rebase.

**Comment by nicain** [comment]
@dorukozturk These last two checks, travis and appveyor, seem to take forever.  appveyor never even ran on my last PR.  Any thoughts about what the slow-down could be?  Maybe our project got throttled?

**Comment by nicain** [comment]
@dorukozturk ping.  This delay in running ci seems really strange.  Any ideas what could be going on?

**Comment by dorukozturk** [comment]
@nicain Travis was always slow. I think they cannot run macOS virtually so there is a big queue. Appveyor is weird though. I think I will just contact them and see what is going on. If @jcfr thinks it is ok we can maybe merge this PR since circle tests pass. I don't want to block you anymore.

**Comment by nicain** [comment]
OK.  TBH I am blocked by the fix you provide here, and the slow CI from travis and appveyor, so even if you do merge this Ill still need to wait until we can resolve the travis/appveyor slowness.  I need the change in #288 to keep working on #275, so I can just keep driving at that.  So my opinion is don't merge, lets get to the root of the problem.

**Comment by dorukozturk** [comment]
@nicain Sorry I thought I was commenting on your PR as opposed to mine (I mean maybe we should merge your PR) . I contacted the Appveyor people and they fixed the Windows build issue. Travis is still up in the air and I am trying to fix it.

**Comment by nicain** [comment]
Awesome.  Just saw this: https://circleci.com/blog/one-more-thing-apple-developers-can-now-build-for-macos-ios-tvos-and-watchos-on-circleci-2-0/ We are using travis for macOS, right?  Can we move that over to circleci?

**Comment by dorukozturk** [comment]
@nicain We just talked about this with Jc. I think it is a paid option but it should be a big gain time-wise on our builds.

**Comment by jcfr** [comment]
> circleci 2.0 macos

I tried to set CircleCI 2 for macos (in addition of the docker) up for a project with the same problem. See https://github.com/scikit-build/scikit-ci-addons/pull/64/commits/7f0c8d58bd87b969b408905c39ba732fd98d85f9

But it is shown as "Not running", this happen because it is not free.

Here are the pricing details: https://circleci.com/pricing/#build-os-x

Cc: @mgrauer 


That said, I will have an other look at the pip issue on travis for macos. This happen because of the recent update of macOS image by Travis and the fact we try to install the support tooling on system python.

File: .travis.yml
Comment (jcfr) [inline .travis.yml]: The following will work:

```
curl --silent --show-error https://bootstrap.pypa.io/get-pip.py | sudo -H python
```

**Review by nicain — DISMISSED** [review]
Thanks all, glad this seems to be working.

File: .circleci/config.yml
Comment (jcfr) [inline .circleci/config.yml]: This should always be done.

Comment (jcfr) [reply .circleci/config.yml]: The variable should probably be named  `UPLOAD_WHEELS` or `PUBLISH_WHEELS` ... having the word `RELEASE` in the name of the variable seems that the actual "stable release" is published ...

Comment (nicain) [reply .circleci/config.yml]: Yup I agree; this tricked me until you made this comment.

Comment (dorukozturk) [reply .circleci/config.yml]: Ok, I rebased it (changes in the cirlcle config are gone for this PR) and will merge this since this PR fixed the travis issue.

Comment (jcfr) [reply .circleci/config.yml]: Full disclosure, I made the commit adding this variable. I will do a follow up Pr as soon as CI fixes are all integrated

**Pull request merged**
